File: CS/RefreshDataExample/ViewModel/RefreshDataViewModel.cs
using DataUpdatesExample;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RefreshDataExample
{
    public class RefreshDataViewModel : UpdateViewModelBase
    {
        protected override void ConfigureSettings() {
            dataSourceSize = 5000;
            groupsCount = dataSourceSize / 30;
            setAllRowValuesOnUpdate = true;
            collectionUpdatesPerSecond = dataSourceSize;
            rowUpdatesPerSecond = dataSourceSize;
        }
        protected override IList<Device> CreateSourceCollection() {
            return new List<Device>();
        }
    }
}

File: VB/RefreshDataExample/ViewModel/RefreshDataViewModel.vb
Imports DataUpdatesExample
Imports System.Collections.Generic

Namespace RefreshDataExample

    Public Class RefreshDataViewModel
        Inherits UpdateViewModelBase

        Protected Overrides Sub ConfigureSettings()
            dataSourceSize = 5000
            groupsCount = dataSourceSize \ 30
            setAllRowValuesOnUpdate = True
            collectionUpdatesPerSecond = dataSourceSize
            rowUpdatesPerSecond = dataSourceSize
        End Sub

        Protected Overrides Function CreateSourceCollection() As IList(Of Device)
            Return New List(Of Device)()
        End Function
    End Class
End Namespace
